lib/Reader/readerir.cpp

```diff
@@ -945,6 +971,118 @@ void GenIR::createSafepointPoll() {
 
 bool GenIR::doTailCallOpt() { return JitContext->Options->DoTailCallOpt; }
 
+void GenIR::pushDebugScope(llvm::Function *F) {
+  LLILCDbgInfo.LexicalBlocks.push_back(LLILCDbgInfo.FnScopeMap[F]);
+}
+
+void GenIR::popDebugScope() {
+  LLILCDbgInfo.LexicalBlocks.pop_back();
+}
+
+// Set the Debug Location for the current instruction
+void GenIR::setDebugLocation(uint32_t CurrOffset, uint32_t IsCall) {
+  DIScope *Scope;
+  if (LLILCDbgInfo.LexicalBlocks.empty())
+    Scope = LLILCDbgInfo.TheCU;
+  else
+    Scope = LLILCDbgInfo.LexicalBlocks.back();
+  DebugLoc Loc = DebugLoc::get(CurrOffset, IsCall, Scope);
+
+  LLVMBuilder->SetCurrentDebugLocation(Loc);
+}
+
+llvm::DISubroutineType *GenIR::createFunctionType(llvm::Function *F,
```

Comment:
We should leave a note somewhere about these dwarf type mappings.

The CLR/EE will never look at them, so (I assume) we have them because if we run on linux and use native debugging then gdb or similar is informed about the debug for the jitted code we can get some information when debugging without using the "standard" xplat approach -- though it is debatable how useful this will be is given that there is no ability to map back to source.

I'm fine with keeping this code around but we should add a note about its purpose and why it is ok for it to be incomplete.